To-Do-app/src/Components/PageScrollingInfinite.tsx

```diff
@@ -0,0 +1,86 @@
+import axios from "axios"
+import React, { useEffect, useState } from "react"
+import InfiniteScroll from "react-infinite-scroll-component"
+
+interface IUser {
+    avatar: string
+    first_name: string
+    last_name: string
+    email: string
+}
+
+const PageScrollingInfinite = () => {
+
+    const Per_Page = 2
+    const apiPath = "https://reqres.in/api/users"
+    const [UserLists, setUserLists] = useState<Array<IUser>>([])
+    const [total, setTotal] = React.useState<Number>(0)
+
+    const getList = () => {
+        const perpage = (UserLists.length / Per_Page) + 1
+        const pagechange = "?page=" + perpage + "&per_page=" + Per_Page
+        const finalUrl = apiPath + pagechange
+
+        axios.get(finalUrl)
+            .then((res) => {
+                const apiRes = res.data.data
+                setTotal(res.data.total)
+                console.log("total", setTotal)
+                const mergeData = [...UserLists, ...apiRes]
+                setUserLists(mergeData)
+            })
+            .catch((err) => {
+                console.error("Loading Error", err)
+            })
+    }
+
+    useEffect(() => {
+        getList()
+    }, [])
+
+    const fetchMoreData = () => {
+        setTimeout(() => getList(), 1000)
```

Comment:
what is the need to call setTimeout ?

Reply:
To load the next list after 1 second, So I have used setTimeout